lib/improver/utilities/cube_checker.py

```diff
@@ -78,6 +78,58 @@ def check_cube_not_float64(cube, fix=False):
                         coord, cube))
 
 
+def check_coord_datatypes(coord, datatype, fix=False, rounding=False):
+    """Check that the coordinate is of the datatype specified. The coordinate
+    will be modified in place, if the fix keyword is set to True.
+
+    Args:
+        coord (iris.coord.DimCoord, iris.coord.AuxCoord):
+            The coordinate that will be checked for its datatype.
+        datatype (type):
+            The datatype that is expected for the coordinate.
+            For example, np.int32, np.float32, np.float64, etc.
+
+    Keyword Args:
+       fix (bool):
+           If fix is True, then the cube is amended to not include datatype
+           specified, otherwise, an error will be raised.
+       rounding (bool):
+           If fix is True, then the rounding keyword can also be enabled,
+           that will round the coordinate points and bounds prior to changing
+           the datatype.
+
+    Raises:
+        TypeError: The coordinate points are not of the expected datatype.
+        TypeError: The coordinate bounds are not of the expected datatype.
+    """
+    if coord.points.dtype != datatype:
+        if fix:
+            if rounding:
+                coord.points = np.around(coord.points)
+            coord.points = coord.points.astype(datatype)
```

Comment:
I'm a bit worried about this function that it doesn't provide any indication (warning or error) if the coordinate values are significantly changed by rounding.  For example if a time coordinate in hours is rounded to int64, but is in fact on a half-hour, the resulting value is wrong.

I think there are two options to solve this.  Either we have a general function that should check both units and datatypes (so that we are sure time units are in seconds before we make them integers).  Or we have a function to deal with time coordinates that is separate from other coordinate types, that casts to seconds first.  The first solution is more general, but the second may be quicker to implement.